pkg/rbac/rbac.go

```diff
@@ -0,0 +1,287 @@
+						// add verbs that are not already in the list
+						resourceAccessResults[ruleResourceName] = addUniqueItems(
+							resourceAccessResults[ruleResourceName], rule.Verbs...)
+					}
+				}
+			} else {
+				// if no resource names are set on the rule, it means acls in it apply to all resources of this type
+				if len(resourcenames) != 0 {
+					// if given resourcenames is nonempty,
+					// add acls to  each of the resourcename as the rule appplies to all  of the type
+					for _, rname := range resourcenames {
+						// add verbs that are not already in the list
+						resourceAccessResults[rname] = addUniqueItems(resourceAccessResults[rname], rule.Verbs...)
+					}
+				} else {
+					// if given resourcenames is empty,  add acls under the "*" entry as rule applied to all
+					resourceAccessResults["*"] = addUniqueItems(resourceAccessResults["*"], rule.Verbs...)
+				}
+			}
+		}
+	}
+
+	// the above only adds entries for the resourcenames that have some acls associated with it
+	// so add empty entries for any resourcenames that are missing in the result list
+	if len(resourcenames) != 0 {
+		for _, rname := range resourcenames {
+			if _, ok := resourceAccessResults[rname]; !ok {
+				resourceAccessResults[rname] = []string{}
+			}
+		}
+	}
+
+	klog.V(2).Infof("Resource access results %v", resourceAccessResults)
+
+	return resourceAccessResults, nil
+}
+
+// a convenience method for building a list with unique entries
+// specified items are added to the given  slice  of items if not already in it
+
+func addUniqueItems(itemlist []string, itemsToAdd ...string) []string {
+	klog.V(2).Infof("Adding items %v to list %v", itemsToAdd, itemlist)
+
+	// iterate through each of the items in  itemsToAdd list
+	// and add item only  if it doesnt already exist in the itemlist
+	resultList := itemlist
+	for _, item := range itemsToAdd {
+		// check if itemList  contains the item before appending
+		if !slices.Contains(resultList, item) {
+			resultList = append(resultList, item)
+		}
+	}
+
+	klog.V(2).Infof("Result List: %v ", resultList)
+
+	return resultList
+}
+
+// makes a selfsubjectrulesreview call , the call requires a namespace to be set
+// call returns the list of the allowed acls for the user
+// for namespace-scoped resources in a given namespace +  acls for  all cluster-scoped resources
+// if no namespace is set, e.g. if the user only cares about the acls for cluster scoped resources,
+// then the namespace defaults to  "default"
+// When the user's bearer token is used to make the call instead of  impersonation,
+// the user's usergroup are already taken into account when returning the list of acls
+// if the call is successful then this function returns a list of resourcerules
+func MakeSubjectAccessRulesReviewForUser(
+	kclient kubernetes.Interface, namespace string,
+) ([]authorizationv1.ResourceRule, error) {
+	klog.V(2).Infof("Make Subject Access Rules Review for Namespace %s", namespace)
+
+	if len(namespace) == 0 {
```

Comment:
Optional: This is the more common approach in Go
```suggestion
	if namespace == "" {
```